Core/include/Acts/Vertexing/HelicalTrackLinearizer.ipp

```diff
@@ -19,17 +19,21 @@ Acts::Result<Acts::LinearizedTrack> Acts::
   const std::shared_ptr<PerigeeSurface> perigeeSurface =
       Surface::makeShared<PerigeeSurface>(linPointPos);
 
+  auto intersection = perigeeSurface->intersect(gctx, params.position(gctx),
+                                                params.unitDirection(), false);
+
   // Create propagator options
   auto logger = getDefaultLogger("HelTrkLinProp", Logging::INFO);
   propagator_options_t pOptions(gctx, mctx, LoggerWrapper{*logger});
-  pOptions.direction = NavigationDirection::Backward;
+  pOptions.direction = intersection.intersection.pathLength >= 0
+                           ? NavigationDirection::Forward
+                           : NavigationDirection::Backward;
```

Comment:
So we only to the one propagation in this case then? Or should we failsafe it by trying both but in this ordering?

Reply:
I tried to do both directions here https://github.com/acts-project/acts/pull/1363/commits/4cc26a2966ac381905bf204d6a598b5684368409 just without order

I think we can go with the single direction for now and if that does not work at some point we could think about having an any direction propagation